spotbugs/src/main/java/edu/umd/cs/findbugs/detect/PublicIdentifiers.java

```diff
@@ -0,0 +1,11451 @@
+/*
+ * SpotBugs - Find bugs in Java programs
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ */
+
+package edu.umd.cs.findbugs.detect;
+
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * This class contains all the public identifiers that are part of the
+ * Java Standard Library. It is used to detect any shadowing or obscuring
+ * of these identifiers.
+ * Since there are so many identifiers, they are loaded from separate
+ * functions to avoid compiler limits. The functions load roughly the same number
+ * of identifiers.
+ */
+public class PublicIdentifiers {
+    private static final Set<String> STANDARD_LIBRARY_PACKAGES = new HashSet<>();
+    public static final Set<String> PUBLIC_IDENTIFIERS = new HashSet<>();
+
+    static {
+        addSTANDARD_PACKAGES();
+        addPUBLIC_IDENTIFIERSAB();
+        addPUBLIC_IDENTIFIERSCC();
+        addPUBLIC_IDENTIFIERSDE();
+        addPUBLIC_IDENTIFIERSFH();
+        addPUBLIC_IDENTIFIERSIJ();
+        addPUBLIC_IDENTIFIERSKM();
+        addPUBLIC_IDENTIFIERSNR();
+        addPUBLIC_IDENTIFIERSSS();
+        addPUBLIC_IDENTIFIERSTU();
+        addPUBLIC_IDENTIFIERSVZ();
+    }
+
+    public static boolean isPartOfStandardLibrary(String containingPackage) {
+        return STANDARD_LIBRARY_PACKAGES.stream().anyMatch(containingPackage::startsWith);
+    }
+
+    private static void addSTANDARD_PACKAGES() {
```

Comment:
What is going on with the capitalisation of this name?

Reply:
@ThrawnCA This has been updated. It was some strange idea of mine and a mistake.